server/src/main/java/org/elasticsearch/health/metadata/HealthMetadataService.java

```diff
@@ -102,63 +106,43 @@ private void registerListeners() {
             .forEach(
                 setting -> clusterSettings.addSettingsUpdateConsumer(
                     setting,
-                    value -> updateOnSettingsUpdated(new UpdateDiskMetadata(setting.getKey(), value.getStringRep()))
+                    value -> updateOnDiskSettingsUpdated(setting.getKey(), value.getStringRep())
                 )
             );
 
         Stream.of(SETTING_CLUSTER_MAX_SHARDS_PER_NODE, SETTING_CLUSTER_MAX_SHARDS_PER_NODE_FROZEN)
             .forEach(
                 setting -> clusterSettings.addSettingsUpdateConsumer(
                     setting,
-                    value -> updateOnSettingsUpdated(new UpdateShardLimitsMetadata(setting.getKey(), value))
+                    value -> updateOnShardLimitsSettingsUpdated(setting.getKey(), value)
                 )
             );
-
-        clusterService.getClusterSettings().addSettingsUpdateConsumer(ENABLED_SETTING, this::enable);
     }
 
-    private void enable(boolean enabled) {
+    private void updateOnHealthNodeEnabledChange(boolean enabled) {
         this.enabled = enabled;
         if (this.enabled) {
             clusterService.addListener(clusterStateListener);
-            resetHealthMetadata("health-node-enabled");
+            taskQueue.submitTask("health-node-enabled", () -> this.localHealthMetadata, null);
```

Comment:
Shall we also wrap this call in 
```
           if (this.localHealthMetadata.equals(HealthMetadata.getFromClusterState(clusterService.state())) == false) {
              ...           
           }
```
?

Reply:
Talking of `isMaster` - do we have a small bug here were this would submit a cluster state change even if this particular node is not the master? 🤔 
I reckon we should also check here if we are the master node before sending this update? (in which case we probably still need the `isMaster` dance)

What do you think?

Reply:
you're right, `isMaster` was transitively check through `readyToPublish`, good catch

Reply:
done